doc/cygnus-ngsi/flume_extensions_catalogue/ngsi_sth_sink.md

```diff
@@ -47,15 +54,57 @@ This is done at the Cygnus Http listeners (in Flume jergon, sources) thanks to [
+
+It must be said [MongoDB does not accept](https://docs.mongodb.com/manual/reference/limits/#naming-restrictions) `$` in the collection names, so it will be replaced by underscore, `_`.
+
+The following table summarizes the table name composition (assuming default `sth_` prefix):
+
+| FIWARE service path | `dm-by-service-path` | `dm-by-entity` | `dm-by-attribute` |
+|---|---|---|---|
+| `/` | `sth_/.aggr` | `sth_/<entityId>_<entityType>.aggr` | `sth_/<entityId>_<entityType>_<attrName>.aggr` |
+| `/<svcPath>.aggr` | `sth_/<svcPath>.aggr` | `sth_/<svcPath>_<entityId>_<entityType>.aggr` | `sth_/<svcPath>_<entityId>_<entityType>_<attrName>.aggr` |
 
-The context attributes within each context response/entity are iterated, and a new Json document is appended to the current collection.
+Please observe the concatenation of entity ID and type is already given in the `notified_entities`/`grouped_entities` header values (depending on using or not the grouping rules, see the [Configuration](#section2.1) section for more details) within the Flume event.
+
+[Top](#top)
+
+####<a name="section1.2.3"></a>Storing
+As said, `NGSISTHSink` has been designed for pre-aggregating certain statistics about entities and their attributes:
+
+* For numeric attribute values:
+    * Sum of all the samples.
+    * Sum of the square value of all the samples.
+    * Maximum value among all the samples.
+    * Minimum value among all the samples.
+* Number of occurrences for string attribute values.
+
+This is done by changing the <i>usual</i> behaviour of a NGSI-like sink: instead of appending more and more information elements, a set of information elements (in this case, Json documents within the collections) are created once, and updated many.
+
+There will be at least (see below about the <i>origin</i>) a Json document for each <i>resolution</i> handled by `NGSISTHSink`: month, day, hour, minute and second. For each one of these documents, there will be as many <i>offset</i> fields as the resolution denotes, e.g. for a resolution of "day" ther will be 24 offsets, for a resolution of "second" there will be 60 offsets. Initially, these offsets will be setup to 0 (for numeric aggregations), or `{}` for string aggregations, and as long notifications arrive, these values will be updated depending on the resolution and the offset within that resolution the notification was receieved.
+
+Each one of the Json documents (each one for each resolution) will also have an <i>origin</i>, the time for which the aggregated information applies. For instance, for a resolution of "minute" a valid origin could be `2015-03-01T13:00:00.000Z`, meaning the 13th hour of March, the 3rd, 2015. There may be another Json document having a differetn origin for the same resolution, e.g. `2015-03-01T14:00:00.000Z`, meaning the 14th hour of March, the 3rd, 2015. The origin is stored using UTC time to avoid locale issues.
```

Comment:
Fixed in https://github.com/telefonicaid/fiware-cygnus/pull/1063/commits/445b0f5188cad2ea6997d1a20ca1c13150d076a5